### PR TITLE
Refactoring salvo processor

### PR DESCRIPTION
**Pull request by szczepanskikrs**
Added tests and refactor to server.

### CODE REVIEW

File: common/src/main/java/battleships/communication/messages/Salvo.java [hunk 1]
```diff
@@ -2,16 +2,35 @@
...
+    public static Salvo createForPositions(Integer... positions) {
+        return new Salvo(Arrays.stream(positions).collect(Collectors.toList()));
+    }
+
+
     public Salvo(List<Integer> salvoPositions) {
         this.salvoPositions = salvoPositions;
     }
 
+    public void setSalvoPositions(List<Integer> salvoPositions) {
```
Comment (robwrobel) [inline common/src/main/java/battleships/communication/messages/Salvo.java]: it is not used

File: common/src/main/java/battleships/ships/Ship.java [hunk 2]
```diff
@@ -3,13 +3,22 @@
...
 
-public class Ship implements Messagable{
+public class Ship implements Messagable {
     private List<Mast> masts;
 
-    public Ship() {}
+    public Ship() {
+    }
+
+    public static Ship viaList(List<Integer> positions) {
```
Comment (robwrobel) [inline common/src/main/java/battleships/ships/Ship.java]: do we need both viaList and createShip methods?

File: client/src/main/java/battleships/game/Board.java [hunk 3]
```diff
@@ -4,7 +4,9 @@
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
-class Board {
+public class Board {
+
+    //todo why not board with one row before and after border??
```
Comment (igasm) [inline client/src/main/java/battleships/game/Board.java]: please, remove todo

File: common/src/main/java/battleships/communication/messages/Salvo.java [hunk 4]
```diff
@@ -2,16 +2,35 @@
...
 
-public class Salvo implements Messagable{
+public class Salvo implements Messagable {
     private List<Integer> salvoPositions;
 
+    public Salvo() {
+    }
+
+    public static Salvo createForPositions(Integer... positions) {
+        return new Salvo(Arrays.stream(positions).collect(Collectors.toList()));
```
Comment (igasm) [inline common/src/main/java/battleships/communication/messages/Salvo.java]: What is that for? Is this method called anywhere else then in tests? Becasue if only there, this is not a good idea.

File: common/src/main/java/battleships/communication/messages/SalvoResult.java [hunk 5]
```diff
@@ -6,15 +6,26 @@
...
     public List<Integer> getResultList() {
         return resultList;
     }
+
+    public List<Integer> getSalvoPositions() {
+        return resultList;
+    }
+
+    @Override
+    public String toString() {
```
Comment (igasm) [inline common/src/main/java/battleships/communication/messages/SalvoResult.java]: With this toString method, I can't really tell which part of String belongs to resultList and which belongs to salvoPositions.

File: common/src/main/java/battleships/ships/Mast.java [hunk 6]
```diff
@@ -32,4 +32,8 @@ public Integer getPosition() {
     void kill() {
         isAlive = false;
     }
+
+    public String toString(){
```
Comment (igasm) [inline common/src/main/java/battleships/ships/Mast.java]: where is @override annotation?

File: common/src/main/java/battleships/ships/Ship.java [hunk 7]
```diff
@@ -3,13 +3,22 @@
...
+        List<Mast> masts = new ArrayList<>();
+        positions.forEach(p -> masts.add(new Mast(p)));
+        return new Ship(masts);
+    }
 
     public static Ship createShip(Integer... positions) {
```
Comment (igasm) [inline common/src/main/java/battleships/ships/Ship.java]: I don't like it. As before: what is that for? Is this method called anywhere else then in tests? Becasue if only there, this is not a good idea.

File: server/src/main/java/battleships/gameplay/SalvoProcessor.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        List<SalvoResult> salvoResults = new ArrayList<>();
+
+        IntStream.range(0, fleets.size())
+                .forEach(p -> salvoResults.add(calculate(salvos.get(p), fleets.get(p))));
+
+        return salvoResults;
+    }
+
+    private SalvoResult calculate(Salvo salvo, Fleet fleet) {
+        List<Integer> resultingPositions = new ArrayList<>(salvo.getSalvoPositions());
```
Comment (robwrobel) [inline server/src/main/java/battleships/gameplay/SalvoProcessor.java]: here is a bug to correct, calculate function should work on both salvos to work correctly

File: server/src/test/java/battleships/gameplay/SalvoProcessorTest.java [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+    @DataProvider(name = "provider")
+    public static Object[][] provider() {
+        return new Object[][]{
+                {new Quadruplet(
+                        Salvo.createForPositions(1, 2, 3, 4, 5),
+                        Salvo.createForPositions(5, 6, 7),
+                        new Fleet(Collections.singletonList(Ship.createShip(3, 5, 4))),
+                        new Fleet(Collections.singletonList(Ship.createShip(1, 2, 3)))),
+                        Arrays.asList(1, 2, 3), 1},
```
Comment (robwrobel) [inline server/src/test/java/battleships/gameplay/SalvoProcessorTest.java]: it looks there is a bug here it, should be Arrays.asList(1,2,3), 0 }

File: server/src/test/java/battleships/gameplay/SalvoProcessorTest.java [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+                {new Quadruplet(
+                        Salvo.createForPositions(59),
+                        Salvo.createForPositions(7, 8),
+                        new Fleet(Collections.singletonList(Ship.createShip(3, 5, 4))),
+                        new Fleet(Collections.singletonList(Ship.createShip(1, 2, 3, 4, 5, 59)))),
+                        Collections.singletonList(59), 1},
```
Comment (robwrobel) [inline server/src/test/java/battleships/gameplay/SalvoProcessorTest.java]: similar error as above, should be Collections.singletonList(59), 0 }

File: server/src/test/java/battleships/gameplay/SinkingShipsTest.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
+package battleships.gameplay;
+
+import org.testng.annotations.Test;
+
+public class SinkingShipsTest {
+
+    @Test
+    public void testName() {
+        //TODO: REFACTOR and PARAMETRIZE!
+//        List<Fleet> test = new ArrayList<>();
```
Comment (robwrobel) [inline server/src/test/java/battleships/gameplay/SinkingShipsTest.java]: remove this test ?

File: common/src/main/java/battleships/communication/messages/Salvo.java (hunk 1 above)
Comment (szczepanskikrs) [reply common/src/main/java/battleships/communication/messages/Salvo.java]: It is needed for JSON to work correctly

File: common/src/main/java/battleships/ships/Ship.java (hunk 2 above)
Comment (szczepanskikrs) [reply common/src/main/java/battleships/ships/Ship.java]: yes we need them for client to work and for tests to not look like shit :D

**Pull request merged**